dpnp/dpnp_algo/dpnp_algo_mathematical.pyx

```diff
@@ -344,6 +320,28 @@ cpdef dparray dpnp_multiply(dparray x1, x2):
         return call_fptr_2in_1out(DPNP_FN_MULTIPLY, x1, x2, x1.shape)
 
 
+cpdef dparray dpnp_nancumprod(dparray x1):
+
+    cur_x1 = dpnp.copy(x1)
```

Comment:
You can check in one cycle if given array has `nan` element, if exists then make a copy.